src/mongo_odm/test/query_builder.cpp

```diff
@@ -0,0 +1,137 @@
+
+#include <mongo_odm/odm_collection.hpp>
+#include <mongo_odm/query_builder.hpp>
+
+using namespace mongocxx;
+using namespace mongo_odm;
+
+class Bar {
+   public:
+    Bar *v;
+    int64_t w;
+    int x1;
+    int x2;
+    bool y;
+    std::string z;
+    ODM_MAKE_KEYS(Bar, NVP(v), NVP(w), NVP(x1), NVP(x2), NVP(y), NVP(z))
+
+    template <class Archive>
+    void serialize(Archive &ar) {
+        ar(CEREAL_NVP(w), CEREAL_NVP(x1), CEREAL_NVP(x2), CEREAL_NVP(y), CEREAL_NVP(z));
+    }
+};
+ODM_MAKE_KEYS_STORAGE(Bar);
+
+TEST_CASE("Query Builder", "[mongo_odm::query_builder]") {
+    instance::current();
+    client conn{uri{}};
+    collection coll = conn["testdb"]["testcollection"];
+    odm_collection<Bar> bar_coll(coll);
+    coll.delete_many({});
+
+    bar_coll.insert_one({nullptr, 444, 1, 2, false, "hello"});
+    bar_coll.insert_one({nullptr, 444, 1, 3, false, "hello"});
+    bar_coll.insert_one({nullptr, 555, 10, 2, true, "goodbye"});
+
+    SECTION("Test == comparison.", "[mongo_odm::ComparisonExpr]") {
+        auto res = bar_coll.find_one(ODM_KEY(Bar::x1) == 1);
+        REQUIRE(res);
+        REQUIRE(res.value().x1 == 1);
+
+        res = bar_coll.find_one(ODM_KEY(Bar::z) == "hello");
+        REQUIRE(res);
+        REQUIRE(res.value().z == "hello");
+    }
+
+    SECTION("Test > comparison.", "[mongo_odm::ComparisonExpr]") {
+        auto res = bar_coll.find_one(ODM_KEY(Bar::x1) > 1);
+        REQUIRE(res);
+        REQUIRE(res.value().x1 > 1);
+    }
+
+    SECTION("Test >= comparison.", "[mongo_odm::ComparisonExpr]") {
+        auto res = bar_coll.find_one(ODM_KEY(Bar::x1) >= 10);
+        REQUIRE(res);
+        REQUIRE(res.value().x1 >= 10);
+    }
+
+    SECTION("Test < comparison.", "[mongo_odm::ComparisonExpr]") {
+        auto res = bar_coll.find_one(ODM_KEY(Bar::x1) < 10);
+        REQUIRE(res);
+        REQUIRE(res.value().x1 < 10);
+    }
+
+    SECTION("Test <= comparison.", "[mongo_odm::ComparisonExpr]") {
+        auto res = bar_coll.find_one(ODM_KEY(Bar::x1) <= 1);
+        REQUIRE(res);
+        REQUIRE(res.value().x1 <= 1);
+    }
+
+    SECTION("Test != comparison.", "[mongo_odm::ComparisonExpr]") {
+        auto res = bar_coll.find_one(ODM_KEY(Bar::x1) != 1);
+        REQUIRE(res);
+        REQUIRE(res.value().x1 != 1);
+
+        res = bar_coll.find_one(ODM_KEY(Bar::z) != "hello");
+        REQUIRE(res);
+        REQUIRE(res.value().z == "goodbye");
+    }
+
+    SECTION("Test $not expression, with operator!.", "[mongo_odm::NotExpr]") {
+        auto res = bar_coll.find_one(!(ODM_KEY(Bar::x1) < 10));
+        REQUIRE(res);
+        REQUIRE(res.value().x1 >= 10);
+
+        res = bar_coll.find_one(!(ODM_KEY(Bar::z) == "hello"));
+        REQUIRE(res);
+        REQUIRE(res.value().z == "goodbye");
+    }
+
+    SECTION("Test expression list.", "[mongo_odm::ExpressionList]") {
+        auto res = bar_coll.find_one(
+            (ODM_KEY(Bar::x1) == 1, ODM_KEY(Bar::x2) == 2, ODM_KEY(Bar::w) >= 444));
```

Comment:
Otherwise, the expressions get interpreted as separate arguments to `find_one()`